src/main.rs

```diff
@@ -164,19 +193,23 @@ fn handle_suggestions(
         let mut i = 0;
         for solution in suggestion.solutions.iter() {
             println!("\n{}", solution.message);
+
+            // check whether we can squash all suggestions into a list
+            if solution.replacements.len() > 1 {
+                let first = solution.replacements[0].clone();
+                if solution.replacements.iter().all(|s| first.snippet.file_name == s.snippet.file_name && first.snippet.line_range == s.snippet.line_range) {
```

Comment:
Can you put this in a `let all_suggestion_are_alternatives_to_each_other = …` (or something shorter)? Then, this line becomes the much more readable `if all_suggestion_are_alternatives_to_each_other {`.